crates/transaction-pool/src/maintain.rs

```diff
@@ -64,17 +95,100 @@ where
     // keeps track of the state of the pool wrt to blocks
     let mut maintained_state = MaintainedPoolState::InSync;
 
+    // the future that reloads accounts from state
+    let mut reload_accounts_fut = Fuse::terminated();
+
+    // The update loop that waits for new blocks and reorgs and performs pool updated
     // Listen for new chain events and derive the update action for the pool
     loop {
+        trace!(target = "txpool", state=?maintained_state, "awaiting new block or reorg");
+
         metrics.set_dirty_accounts_len(dirty_addresses.len());
+        let pool_info = pool.block_info();
 
-        let Some(event) = events.next().await else { break };
+        // after performing a pool update after a new block we have some time to properly update
+        // dirty accounts and correct if the pool drifted from current state, for example after
+        // restart or a pipeline run
+        if maintained_state.is_drifted() {
+            // assuming all senders are dirty
+            dirty_addresses = pool.unique_senders();
+            maintained_state = MaintainedPoolState::InSync;
```

Comment:
is this right? i would expect that we update the maintained state at the end of the function, but maybe it doesn't matter here?

Reply:
drifted is set after new canon chain emit after that the pool is in sync again but (hash, basefee updated etc) but we now need to update accounts again